.github/workflows/test-sui.yaml

```diff
@@ -0,0 +1,148 @@
+name: Test Sui
+
+on: pull_request
+
+jobs:
+  test-sui:
+    runs-on: blacksmith-2vcpu-ubuntu-2204
+
+    steps:
```

Comment:
Since this is a slower action, we could make it only run when there's a change to `sui/` or `common/`